### PR TITLE
Add Deepseek V4 Flash E2E changes to nightly CI

### LINKED ISSUE #4742: Add Deepseek v4 to CI
Need to add the following modified DSv4 model tests to tt-xla CI to catch regressions:
On LLMBox:
- Attention tests (Prefill and Decode)
- Flash MoE (hash and non-hash layers)
- Block Prefill and Decode (single layer including MoE and Attention)
- Transformer Prefill and Decode (1, 2, and 3 layers)

On BH Galaxy:
- Full E2E coherence test

### PR DESCRIPTION
**Pull request by hshahTT**
### Ticket
Fixes https://github.com/tenstorrent/tt-xla/issues/4742

### What's changed
Adding an E2E test for DSv4 Flash that runs a full prefill + 10 decode iterations and checks the logits PCC > 0.95 for each step.

### Checklist
- [x] New/Existing tests provide coverage for changes

### CODE REVIEW

**Comment by hshahTT** [comment]
Note: I changed the nightly json file to remove all tests except for the BH galaxy one. This was done so that I can run just the BH galaxy test in CI without wasting resources spinning up other jobs. I will add back those tests before submitting this PR.

**Review by vvukomanTT — APPROVED** [review]
Changes look good, thanks!
(reminder to revert the removal of existing basic nightly tests 👍 )

**Review by AleksKnezevic — APPROVED** [review]

File: .github/workflows/test-matrix-presets/basic-test-nightly.json
Comment (jazpurTT) [inline .github/workflows/test-matrix-presets/basic-test-nightly.json:1]: I assume this was for testing, but this should just add the galaxy-bh line to the already existing tests, or am I missing something?

File: tests/torch/models/deepseek_v4/test_deepseek_v4_e2e.py [hunk 1]
```diff
@@ -0,0 +1,344 @@
...
+            compound = ("batch", "model")
+            shard_specs[experts.gate_proj] = (compound, None, None)
+            shard_specs[experts.up_proj] = (compound, None, None)
+            shard_specs[experts.down_proj] = (compound, None, None)
+    return shard_specs
+
+
+@pytest.mark.nightly
+@pytest.mark.bh_galaxy
+@pytest.mark.skip(
```
Comment (gengelageTT) [inline tests/torch/models/deepseek_v4/test_deepseek_v4_e2e.py]: Is this test supposed to be skipped? Looks like copilot made this change https://github.com/tenstorrent/tt-xla/pull/4841/commits/229414b673863e7ea36c80eccf256bbf2381f91a

Comment (hshahTT) [reply tests/torch/models/deepseek_v4/test_deepseek_v4_e2e.py]: good catch. I have no idea why it decided to do that. Will revert

File: .github/workflows/test-matrix-presets/basic-test-nightly.json
Comment (hshahTT) [reply .github/workflows/test-matrix-presets/basic-test-nightly.json:1]: Yeah I'll re-add these tests before submitting. Just removed them so that when I kick of a nightly it doesnt waste other resources.
